LCAPI.Core/ModData.cs

```diff
@@ -0,0 +1,186 @@
+{
+    internal static ES3Settings defaultSetting;
+    internal static HashSet<string> tracker = new();
+    internal static void Setup()
+    {
+        defaultSetting = new ES3Settings
+        {
+            location = ES3.Location.File,
+            prettyPrint = true,
+            saveChildren = true,
+            compressionType = ES3.CompressionType.Gzip,
+            bufferSize = 4096,
+            encoding = Encoding.UTF8
+        };
+    }
+
+    internal void SetupConfig()
+    {
+        if (useCustomPath)
+        {
+            config = new($"{customPath}.sav", defaultSetting);
+            return;
+        }
+
+        config = new($"{guid}_LCModData.sav", defaultSetting);
+    }
+
+    internal string guid;
+    internal string customPath;
+    internal bool useCustomPath;
+    internal ES3Settings config;
+    internal bool blocked = false;
+
+    /// <summary>
+    /// Create an instance of ModData using your GUID identifier to prevent key duplication.
+    /// </summary>
+    /// <param name="guid"></param>
+    public ModData(string guid)
+    {
+        this.guid = guid;
+        this.customPath = string.Empty;
+
+        if (!tracker.Add(guid))
+        {
+            blocked = true;
+            Plugin.Log.LogError($"ModData: GUID {guid} is already used by other plugins/mods!");
+            throw new ArgumentException($"GUID is already used by other plugins/mods!", nameof(guid));
+        }
+
+        SetupConfig();
+    }
+
+    /// <summary>
+    /// Create an instance of ModData using your GUID identifier to prevent key duplication.<br/>
+    /// Use provided path instead of the default path.
+    /// </summary>
+    /// <param name="guid"></param>
+    /// <param name="path"></param>
+    public ModData(string guid, string path)
+    {
+        this.guid = guid;
+        this.customPath = path;
+        useCustomPath = true;
+
+        SetupConfig();
+    }
+
+    /// <summary>
+    /// Create an instance of ModData using your GUID identifier to prevent key duplication.<br/>
+    /// Use the provided path instead of the default path.<br/>
+    /// Use encryption to store the data.
+    /// </summary>
+    /// <param name="guid"></param>
+    /// <param name="path"></param>
+    /// <param name="encryptionPassword"></param>
+    public ModData(string guid, string path, string encryptionPassword)
+    {
+        this.guid = guid;
+        this.customPath = path;
+        useCustomPath = true;
+        SetupConfig();
+
+        this.config.encryptionType = ES3.EncryptionType.AES;
+        this.config.encryptionPassword = encryptionPassword;
+    }
+
+    /// <summary>
+    /// Enable your mod data to use encryption. (You must use a custom path to have this working)
+    /// </summary>
+    /// <param name="encryptionKey"></param>
+    public void UseEncryption(string encryptionKey)
+    {
+        if (!useCustomPath)
+        {
+            Plugin.Log.LogWarning("ModData: Trying to use encryption without custom path!");
```

Comment:
Why is a custom path required for encryption?

Reply:
My brain cell got burned out and some how this is not deleted, it should now be 'fixed' in my latest commit.